apps/ledger-live-desktop/src/renderer/components/SignMessageConfirm/index.jsx

```diff
@@ -55,43 +59,68 @@ type Props = {
   signMessageRequested: TypedMessageData | MessageData,
```

Comment:
just curious: what could cause the message to be one or the other ?

Reply:
This modal can be triggered from different flows, and it this case it's coming from the Wallet API or through WalletConnect, which can try to validate 2 types of messages: [EIP-191](https://eips.ethereum.org/EIPS/eip-191) and [EIP-712](https://eips.ethereum.org/EIPS/eip-712). One is just a string, the other is a full standardized JSON object. 

Since what is received through our API or through WalletConnect can be stringified, it means (in a very simplified way) that receiving an object would still be a string in the end, making a `TypedMessageData` looked like a `MessageData` basically.

We try to parse every message received in that flow as a JSON object, which means we can't know the type of message we'll get in the end. If the parsing works, it's going to be an EIP-712 message `TypedMessageData` otherwise it's going to be an EIP-191 message `MessageData`.